src/chunk.c

```diff
@@ -2194,34 +2203,130 @@ ts_chunk_show_chunks(PG_FUNCTION_ARGS)
 		else
 			time_type = InvalidOid;
 
+		/* note that arg_types will be the same for all specified "ANY" elements for a given call */
+		arg_type = InvalidOid;
 		if (!PG_ARGISNULL(1))
-			older_than = ts_time_value_from_arg(PG_GETARG_DATUM(1),
-												get_fn_expr_argtype(fcinfo->flinfo, 1),
-												time_type);
+		{
+			arg_type = get_fn_expr_argtype(fcinfo->flinfo, 1);
+			older_than = ts_time_value_from_arg(PG_GETARG_DATUM(1), arg_type, time_type, true);
+			older_newer = true;
+		}
 
 		if (!PG_ARGISNULL(2))
-			newer_than = ts_time_value_from_arg(PG_GETARG_DATUM(2),
-												get_fn_expr_argtype(fcinfo->flinfo, 2),
-												time_type);
+		{
+			arg_type = get_fn_expr_argtype(fcinfo->flinfo, 2);
+			newer_than = ts_time_value_from_arg(PG_GETARG_DATUM(2), arg_type, time_type, true);
+			older_newer = true;
+		}
+
+		/*
+		 * We cannot have a mix of [older_than/newer_than] and [created_before/created_after].
+		 * So, check that first.
+		 */
+
+		if (!PG_ARGISNULL(3))
+		{
+			if (older_newer)
+				ereport(ERROR,
+						(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
+						 errmsg("cannot specify \"older_than\" or \"newer_than\" together with "
+								"\"created_before\""
+								"or \"created_after\""),
+						 errhint("\"older_than\" and/or \"newer_than\" is recommended with "
+								 "\"time\"-like partitioning"
+								 " and  \"created_before\" and/or \"created_after\" is recommended "
+								 "with \"integer\"-like"
+								 " partitioning.")));
+
+			arg_type = get_fn_expr_argtype(fcinfo->flinfo, 3);
+			created_before = ts_time_value_from_arg(PG_GETARG_DATUM(3), arg_type, time_type, false);
+			before_after = true;
+		}
+
+		if (!PG_ARGISNULL(4))
+		{
+			if (older_newer)
+				ereport(ERROR,
+						(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
+						 errmsg("cannot specify \"older_than\" or \"newer_than\" together with "
+								"\"created_before\""
+								"or \"created_after\""),
+						 errhint("\"older_than\" and/or \"newer_than\" is recommended with "
+								 "\"time\"-like partitioning"
+								 " and  \"created_before\" and/or \"created_after\" is recommended "
+								 "with \"integer\"-like"
+								 " partitioning.")));
```

Comment:
Same here.

Reply:
Btw, these messages seem to be repeated in many places in the code. It might make sense to create a macro/define in order to make sure they are the same in case someone later changes the message in one place.

Reply:
Removed